File: Ruby/problem03.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

def problem03(n)
  case n
  when 0
    return []
  when 1
    return [0]
  else
    retlist = [0] * n
    retlist[1] = 1
    for i in 2...n
      retlist[i] = retlist[i - 1] + retlist[i - 2]
    end
    return retlist
  end
end


if __FILE__ == $0
  p problem03(10)
end

File: Python/problem03.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def problem03(n):
    if n < 1:
        return []
    elif n == 1:
        return [0]
    else:
        retlst = [0, 1]
        for i in range(len(retlst) - 1, n - 1):
            retlst.append(retlst[i] + retlst[i - 1])
        return retlst


if __name__ == '__main__':
    for n in problem03(100):
        print n
